src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -64,6 +64,141 @@ void FieldSpecificationManager::expandObjectCatalogs()
+
+    // Step 1: collect all the set names in a map (this is made necessary by the "apply" loop pattern
+
+    array1d< string > const & setNames = fs.getSetNames();
+    for( localIndex i = 0; i < setNames.size(); ++i )
+    {
+      isTargetSetEmpty[setNames[i]] = 1;
+      isTargetSetCreated[setNames[i]] = 0;
+    }
+
+    // Step 2: apply the boundary condition
+
+    fs.apply< dataRepository::Group >( mesh,
+                                       [&]( FieldSpecificationBase const &,
+                                            string const & setName,
+                                            SortedArrayView< localIndex const > const & targetSet,
+                                            Group & targetGroup,
+                                            string const fieldName )
+    {
+      dataRepository::InputFlags const flag = fs.getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).getInputFlag();
+
+      // 2.a) If we enter this loop, we know that the set has been created
+      //      Fracture/fault sets are created later and the "apply" call silently ignores them
+      isTargetSetCreated.at( setName ) = 1;
+
+      // 2.b) If the fieldName is registered on this target, we record it
+      //      Unfortunately, we need two exceptions:
+      //       - FieldSpecification that do not target a field, like Aquifer, Traction, Equilibrium, etc. For these, the check is not
+      // necessary (the user cannot mess up)
+      //       - Face boundary conditions that target cell-based quantities, like the face BC of the flow solvers
+      if( targetGroup.hasWrapper( fieldName ) ||
+          flag == InputFlags::FALSE || // no need to check input if the input flag is false (Aquifer, Traction, Equilibrium do not target a
+                                       // field)
+          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() ) // the field names of the face BCs are not always
+                                                                                     // registered on
+                                                                                     // the faceManager...
+      {
+        isFieldNameFound = 1;
+      }
+
+      // 2.c) If the target set is not empty, we record it
+      //      Fracture/fault sets are sometimes at initialization and the "apply" call silently ignores them
+      if( targetSet.size() > 0 )
+      {
+        isTargetSetEmpty.at( setName ) = 0;
+      }
+    } );
+
+    // Step 3: MPI synchronization
+
+    isFieldNameFound = MpiWrapper::max( isFieldNameFound );
+
+    for( std::pair< string const, localIndex > & mapEntry : isTargetSetEmpty )
+    {
+      mapEntry.second = MpiWrapper::min( mapEntry.second );
+    }
+    bool areAllSetsEmpty = true;
+    for( std::pair< string const, localIndex > & mapEntry : isTargetSetEmpty )
+    {
+      if( mapEntry.second == 0 ) // target set is not empty
+      {
+        areAllSetsEmpty = false;
```

Comment:
I think you can
```suggestion
        areAllSetsEmpty = false;
        break;
```
but not a big deal.

Reply:
Sounds good, will do. Thanks!

Reply:
Done